syntax = 'proto3';

package image;

service image_tranfer{
    rpc send_me_image(image_request) returns (stream image_response);
    rpc upload_image(stream image_up) returns (ack_response);
    rpc ack(ack_request) returns (ack_response);
}
message image{
    bytes data = 1;
}
message image_request{
    string video = 1;
    string model = 2;
}
message image_response{
    image image_sent = 1;
}
message image_up{
    image image = 1;
}
message ack_request{
    string req = 1;
}
message ack_response{
    string rep = 1;
}
